From 06da6052306d9eb1c136a108550f71ccecf9edb9 Mon Sep 17 00:00:00 2001
From: Ajay Kaher <ajay.kaher@broadcom.com>
Date: Tue, 21 Apr 2026 16:08:33 +0000
Subject: [PATCH] mm: add provision to disable CONFIG_PER_VMA_LOCK

A performance regression has been observed due to PER_VMA_LOCK.

Remove ARCH_SUPPORTS_PER_VMA_LOCK from the X86_64 select config list,
allowing CONFIG_PER_VMA_LOCK to be disabled.

Signed-off-by: Ajay Kaher <ajay.kaher@broadcom.com>

---
 arch/x86/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/x86/Kconfig b/arch/x86/Kconfig
index 36f833c6..5adc8a44 100644
--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -27,7 +27,6 @@ config X86_64
 	# Options that are inherently 64-bit kernel only:
 	select ARCH_HAS_GIGANTIC_PAGE
 	select ARCH_SUPPORTS_INT128 if CC_HAS_INT128
-	select ARCH_SUPPORTS_PER_VMA_LOCK
 	select ARCH_SUPPORTS_HUGE_PFNMAP if TRANSPARENT_HUGEPAGE
 	select HAVE_ARCH_SOFT_DIRTY
 	select MODULES_USE_ELF_RELA
-- 
2.40.4

